### DISCUSSION TITLE
Upgrade guidance from older kube-prometheus?

### DISCUSSION THREAD

**Discussion by Ghazgkull** [prometheus-operator/kube-prometheus#1154] [Q&A]
Is there any guidance available that can help me navigate the upgrade process from kube-prometheus 0.4 to 0.7?

For background, I've inherited ownership of a kube-prometheus deployment which is currently on version 0.4. After upgrading Kubernetes (EKS) from 1.19 to 1.20 yesterday, our AlertManager has started firing constant alerts about not being able to find "all other members of the cluster". Looking at the K8s compatibility matrix for kube-prometheus, I see that 1.20 isn't supported in our ancient version so I'm guessing we just need to upgrade. I know next to nothing about kube-prometheus and I haven't found any documentation on upgrading except for the 0.7 to 0.8 upgrade, which looks like a major undertaking.

**Comment by Kolossi** [comment]
@Ghazgkull : sorry, not an answer but I have exact same issue with an older highly customised `my.jsonnet` file which has the [v0.6 style](https://github.com/prometheus-operator/kube-prometheus/blob/release-0.6/docs/exposing-prometheus-alertmanager-grafana-ingress.md) which seems to have [changed pretty much completely](https://github.com/prometheus-operator/kube-prometheus/commit/b84bd5c72f4d5660538945f3ff0f2eed34941405) going to [v0.7 style](https://github.com/prometheus-operator/kube-prometheus/blob/release-0.7/docs/exposing-prometheus-alertmanager-grafana-ingress.md) but I can't find any guidance at all. Did you get anywhere with this?

From commit histories, @kakkoyun, @paulfantom, @lilic seem to have been involved at least in part in the v0.6->v0.7 changes.  Any chance y'all could give @Ghazgkull and I a steer on this - [migration-guide](https://github.com/prometheus-operator/kube-prometheus/blob/main/docs/migration-guide.md) specifically suggests posting here for assistance, but sadly there's been no reply for a month? Any help would be really appreciated by me :-)

**Reply by Ghazgkull** [reply]
I ended up using this opportunity to finally get off this Jsonnet-based project and moved to the Helm-based kube-promtheus-stack

**Comment by Kolossi** [comment]
Update : See some useful info from @paulfantom in reply at https://kubernetes.slack.com/archives/CFFDS2Z7F/p1624447458459800

**Comment by Kolossi** [comment]
See also PR https://github.com/prometheus-operator/kube-prometheus/pull/1235
